R/data.R

```diff
@@ -522,6 +543,24 @@ spim_lancelot_data_rtm <- function(date, region, model_type, data, full_data,
     data$pillar2_cases_over25 <- data$pillar2_positives_symp_pcr_only_over25
   }
 
+  ## Use symp PCR only for cases by age where available
+  pillar2_symp_PCR_only_by_age <- NULL
+  for (i in pillar2_age_bands) {
+    pillar2_symp_PCR_only_by_age <-
+      rbind(pillar2_symp_PCR_only_by_age,
+            data[, paste0("pillar2_positives_symp_pcr_only_", i)]
+    )
+  }
+  if (!all(is.na(pillar2_symp_PCR_only_by_age))) {
+    if (!full_data) {
+      data$pillar2_cases_over25 <- NA_integer_
+    }
+    for (i in pillar2_age_bands) {
+      data[, paste0("pillar2_cases_", i)] <-
+        data[, paste0("pillar2_positives_symp_pcr_only_", i)]
+    }
```

Comment:
You can do this without a for loop:
```data[, paste0("pillar2_cases_", pillar2_age_bands)] <- data[, paste0("pillar2_positives_symp_pcr_only_", pillar2_age_bands)]

Reply:
Try to avoid for loops unless they are necessary - use vectorisation or apply functions instead where you can